### PR TITLE
Fix infinite loop in `reverse_readlines`

### PR DESCRIPTION
**Pull request by Schamper**
If there's undecodable data in a file, everything before that line is undecodable. `reverse_readllines` would get stuck in an infinite loop if this happened. The fix in `reverse_readlines` breaks out of the loop, but it will miss all lines before the corruption. Perhaps it should raise an error instead?

Also changes the default decoding error handling of `open_decompress` to `backslashreplace` to allow the maximum amount of lines being decoded. With the default of `strict`, you'd lose all lines before corrupt line, instead of just the corrupt line, because of the way Python reads lines, all lines prior to the corrupt byte as well as the first line following the corrupt byte would fail to decode.

While I was at it, I made all text decoding arguments available for pass-thru in `open_decompress`.

### CODE REVIEW

**Comment by Schamper** [comment]
On second thought, changed it to raise an error instead.

File: dissect/target/helpers/utils.py [hunk 1]
```diff
@@ -102,7 +102,13 @@ def year_rollover_helper(
...
                 continue
 
-            relative_ts = datetime.strptime(timestamp.group(0), ts_format)
+            try:
+                relative_ts = datetime.strptime(timestamp.group(0), ts_format)
+            except ValueError as e:
+                log.warning("Timestamp '%s' does not match format '%s', skipping line.", timestamp.group(0), ts_format)
+                log.debug("", exc_info=e)
+                continue
+
```
[the remark this answers is not in the document]
Comment (Schamper) [reply dissect/target/helpers/utils.py:111]: If using a replace based error mode, the line may contain a corrupt timestamp. This is to ensure that we keep on trucking if we encounter a corrupt timestamp.

**Review by pyrco — COMMENTED** [review]
I am correct to understand that if you use the now default `backslashreplace` the `UnicodeDecodeError`'s won't happen anymore as all broken bytes will be backslash encoded?

**Comment by Schamper** [comment]
> I am correct to understand that if you use the now default `backslashreplace` the `UnicodeDecodeError`'s won't happen anymore as all broken bytes will be backslash encoded?

Yes, if you use `open_decompress` (which has somewhat become the standard for Unix log files).

**Review by pyrco — APPROVED** [review]

**Pull request merged**